recipes/_windows.rb

```diff
@@ -0,0 +1,116 @@
+#
+# Cookbook Name:: build-essential
+# Recipe:: _mingw
+#
+# Copyright 2016, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require 'ostruct'
+
+include_recipe '7-zip::default'
+
+def p(url, checksum)
+  OpenStruct.new(url: url, checksum: checksum)
+end
```

Comment:
This should be in a library. I would create `libraries/_helper.rb` and use https://github.com/chef-cookbooks/languages/blob/master/libraries/_helper.rb as a model.